Automate new releases.

zvan opened the pull request:
Gitlab action to automatically publish new version on npm when a new tag is created.

Automation token with name `NPM_TOKEN` was added to project secrets by @davidkuridza 

Failed pipeline can be inspected [here](https://github.com/3fs/eslint-config/actions/runs/1489500970). Pipeline failed because it ran on a commit instead of a tag.

aleks3fs commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,27 @@
+  push:
+    # Limit this action to tags only
+    tags:
+      - '*'
```
Should we limit this to semantic version tags? `v*.*.*`?

zvan replied:
Added as a regex `v[0-9]+\.[0-9]+\.[0-9]+` Tested on private repo and it works

aleks3fs commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,27 @@
+          echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" >> .npmrc
+          RELEASE_VERSION=${GITHUB_REF#refs/*/v}
+          npm --no-git-tag-version version $RELEASE_VERSION
+          npm publish
```
"no new line at end of file"

zvan replied:
🤦 I saw this and then forgot. (fixed)

+1 for we need prettier

aleks3fs reviewed (approved):
Looks good and seems to work.

ZanMarolt commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,27 @@
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
I think we should hardcode the version, so we know if it breaks when updating

```suggestion
    runs-on: ubuntu-20.04
```

zvan replied:
Nice catch thanks. Also changed the job name from` build` to `publish`

JostHren reviewed (approved):
Looks good. 👍

Commit:
Automate new releases.

Gitlab action to automatically publish new version on npm when a new tag is created.
